### PR TITLE
ci: Publish to GitHub packages

### LINKED ISSUE #955: Publish Flank to GitHub Packages
**Author the user story for this feature**

As a Flank developer, I want to publish releases on GitHub Packages so I can have confidence in the release process.

We should publish both snapshot and final releases to GitHub Packages. For now, that means in addition to the release publishing for jcenter and maven central. In the future, we will probably drop jcenter and keep only maven central + GitHub packages.

**Is your feature request related to a problem? Please describe.**

Bintray jcenter is flaky and complex to integrate with. Bintray requires a different account.

**Describe the solution you'd like**

Publish Flank directly to GitHub Packages. Using a first party solution for package management will simplify and standardize the release infrastructure on GitHub.
https://github.com/features/packages

### PR DESCRIPTION
**Pull request by piotradamczyk5**
Fixes #955 

Flank is published to GitHub packages when making a release job

## Checklist

- [x] Documented

### CODE REVIEW

File: test_runner/build.gradle.kts [hunk 1]
```diff
@@ -60,11 +60,21 @@ java {
...
+                password = System.getenv("GITHUB_TOKEN") ?: properties["GITHUB_TOKEN"].toString()
+            }
+        }
+    }
     publications {
         create<MavenPublication>("mavenJava") {
             groupId = "com.github.flank"
             artifactId = artifactID
-            version = System.getenv("MVN_VERSION")
+            version = System.getenv("MVN_VERSION") ?: "local_snapshot"
```
Comment (pawelpasterz) [inline test_runner/build.gradle.kts:77]: If my understanding is correct there is no `MVN_VERSION` if we are publishing snapshot (after the merge to master). How about naming it just `snapshot`? It's not a change request, I'd like just to fully understand it :)

Comment (piotradamczyk5) [reply test_runner/build.gradle.kts:77]: MVN_VERSION is set by CI, `local_snapshot` is used for local builds. Its value is:

- `flank_snapshot` for snapshot build
- `xx.xx.xx` for normal builds

I used `local_snapshot` cause it is also used for printing local version

Comment (pawelpasterz) [reply test_runner/build.gradle.kts:77]: Got it, thanks for the explanation! 👍

**Review by pawelpasterz — APPROVED** [review]

**Comment by piotradamczyk5** [comment]
@Mergifyio rebase

**Review by pawelpasterz — APPROVED** [review]

**Review by Sloox — APPROVED** [review]

**Comment by bootstraponline** [comment]
@Mergifyio refresh

**Pull request merged**